module/zfs/mmp.c

```diff
@@ -217,14 +217,13 @@ mmp_thread_enter(mmp_thread_t *mmp, callb_cpr_t *cpr)
 	mutex_enter(&mmp->mmp_thread_lock);
 }
 
-static _Noreturn void
+static void
```

Comment:
This whole function seems poorly named and pretty pointless after moving `thread_exit()` out of it.

Reply:
Yes, I considered removing it but opted to leave it for another time.  I wouldn't opposed to shedding the `CALLB_CPR_ENTER/CALLB_CPR_EXIT` macros either, on Linux at least these aren't used, and I doubt they're helpful for other platforms since we haven't used them consistently for all other threads.